exasol/analytics/query_handler/graph/stage/sql/dataset.py

```diff
@@ -1,56 +1,12 @@
-import dataclasses
-from enum import Enum
-from typing import Dict, List, Tuple, Union
+from dataclasses import dataclass, field
+from typing import List
 
-from exasol.analytics.query_handler.graph.stage.sql.data_partition import DataPartition
-from exasol.analytics.schema import Column
-from exasol.analytics.utils.data_classes_runtime_type_check import check_dataclass_types
+from exasol.analytics.query_handler.graph.stage.sql.dependency import Dependencies
+from exasol.analytics.schema import Column, TableLike
 
-DataPartitionName = Union[Enum, Tuple[Enum, int]]
-
-
-@dataclasses.dataclass(frozen=True)
+@dataclass(frozen=True)
 class Dataset:
-    """
-    A Dataset consists of multiple data partitions and column lists which indicate the identifier,
-    sample and target columns, The data paritions can be used to describe train and test sets.
-    """
-
-    data_partitions: Dict[DataPartitionName, DataPartition]
+    table_like: TableLike
```

Comment:
This made some tests obsolete.

Reply:
The docsting was also removed.

Reply:
Thanks, I re-added the docstring and updated its content to match the redesign.

Reply:
My take from it:
The `table_like` is the data container. As its type name suggests, the data comes in the form of multiple columns. `identifier_columns` and `columns` are the descriptions of these columns. The meaning of the `dependencies` is unclear, but hopefully the docsting of the `Dependencies` class gives an explanation. If I am wrong please rewrite the docstring.

Reply:
I must admit, that to me the meaning of `dependencies` is also a bit unclear. 

The plural is only defined as a type:

```python
Dependencies = Dict[Enum, Dependency]
```

The Docstring of class `Dependency` has some issues in wording etc., so I would prefer to update it in the scope of the current PR. 

The docstring to me sounds more like a structured comment targeted for human readers rather than to be used by an algorithm.

Reply:
See last push.